mutation_partition.cc

```diff
@@ -2074,19 +2074,22 @@ void query_result_builder::consume(tombstone t) {
 }
 stop_iteration query_result_builder::consume(static_row&& sr, tombstone t, bool is_live) {
     if (!is_live) {
+        _stop = _rb.bump_and_check_tombstone_limit();
         return _stop;
     }
     _stop = _mutation_consumer->consume(std::move(sr), t);
     return _stop;
 }
 stop_iteration query_result_builder::consume(clustering_row&& cr, row_tombstone t,  bool is_live) {
     if (!is_live) {
+        _stop = _rb.bump_and_check_tombstone_limit();
         return _stop;
     }
     _stop = _mutation_consumer->consume(std::move(cr), t);
     return _stop;
 }
 stop_iteration query_result_builder::consume(range_tombstone_change&& rtc) {
+    _stop = _rb.bump_and_check_tombstone_limit();
     return _stop;
 }
```

Comment:
Missing: partition_start with tombstone.

We might want to count individual cell tombstones too, but may be too expensive. To we parse cells here to count their size?

Reply:
I guess is_live will be false for rows that are composed of dead cells only.

Reply:
Right, I missed the partition tombstone.

Rows which have no content are also considered dead by the compactor, and the will be counted as a tombstone by the `is_live` check in `consume(clustering_row)`.

Reply:
> Rows which have no content are also considered dead by the compactor, and the will be counted as a tombstone by the `is_live` check in `consume(clustering_row)`.

I see, makes a lot of sense. If you want to make sure this never gets broken in the future, you can add a test for this specific case (as I suggested above - using UPDATE and DELETE of a single column instead of DELETE of a whole row).